front_end/html/edit_exercise.html

```diff
@@ -848,15 +848,13 @@ <h2>Create exercise</h2>
 
             const container = document.querySelector("#file_upload");
 
-            {% if len(old_tests) > 0 %}
-                var new_tests = {}
+            var new_tests = {}
 
-                {% for test in exercise_details["tests"] %}
-                    new_tests[`{{int(test['test']) - 1}}`] = {'image_output': `{{test['image_output']}}`, 'text_output': `{{test['text_output']}}` }
-                {% end %}
+            {% for i in range(len(exercise_details["tests"])) %}
+                new_tests[`{{i}}`] = {'image_output': `{{exercise_details["tests"][i]['image_output']}}`, 'text_output': `{{exercise_details["tests"][i]['text_output']}}` }
```

Comment:
This reconfiguration addresses a miscellaneous bug I found that arises in a really specific case.